e2e/testcases/cluster_selectors_test.go

```diff
@@ -170,30 +189,42 @@ func TestClusterSelectorOnObjects(t *testing.T) {
 	nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/eng/backend/bob-rolebinding.yaml", rb)
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Change cluster selector to match test cluster")
 	nt.WaitForRepoSyncs()
-	if err := nt.ValidateNotFound(roleBindingName, backendNamespace, &rbacv1.RoleBinding{}); err != nil {
+	if err := nt.ValidateNotFound(rb.Name, rb.Namespace, &rbacv1.RoleBinding{}); err != nil {
 		nt.T.Fatal(err)
 	}
 
 	renameCluster(nt, configMapName, testClusterName)
 	nt.WaitForRepoSyncs()
-	if err := nt.Validate(roleBindingName, backendNamespace, &rbacv1.RoleBinding{}); err != nil {
-		nt.T.Fatal(err)
-	}
+	// TODO: Confirm that the change was Synced.
+	// This is not currently possible using the RootSync status API, because
+	// the commit didn't change, and the commit was already previously Synced.
+	// If sync state could be confirmed, the objects would already be updated,
+	// and we wouldn't need to wait for it.
+	// if err := nt.Validate(rb.Name, rb.Namespace, &rbacv1.RoleBinding{}); err != nil {
+	// 	nt.T.Fatal(err)
+	// }
+	nomostest.WaitForCurrentStatus(nt, kinds.RoleBinding(), rb.Name, rb.Namespace)
 
 	nt.T.Log("Revert cluster selector to match prod cluster")
 	rb.Annotations = inlineProdClusterSelectorAnnotation
 	nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/eng/backend/bob-rolebinding.yaml", rb)
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Revert cluster selector to match prod cluster")
 	nt.WaitForRepoSyncs()
-	if err := nt.ValidateNotFound(roleBindingName, backendNamespace, &rbacv1.RoleBinding{}); err != nil {
+	if err := nt.ValidateNotFound(rb.Name, rb.Namespace, &rbacv1.RoleBinding{}); err != nil {
 		nt.T.Fatal(err)
 	}
 
 	renameCluster(nt, configMapName, prodClusterName)
 	nt.WaitForRepoSyncs()
-	if err := nt.Validate(roleBindingName, backendNamespace, &rbacv1.RoleBinding{}); err != nil {
-		nt.T.Fatal(err)
-	}
+	// TODO: Confirm that the change was Synced.
+	// This is not currently possible using the RootSync status API, because
+	// the commit didn't change, and the commit was already previously Synced.
+	// If sync state could be confirmed, the objects would already be updated,
+	// and we wouldn't need to wait for it.
+	// if err := nt.Validate(rb.Name, rb.Namespace, &rbacv1.RoleBinding{}); err != nil {
+	// 	nt.T.Fatal(err)
+	// }
+	nomostest.WaitForCurrentStatus(nt, kinds.RoleBinding(), rb.Name, rb.Namespace)
```

Comment:
What does `WaitForCurrentStatus` do to help in this context?

Reply:
Wait for existence + validation of current status (reconciled and not terminating). There's a lot of things we could check for, like there's a Predicate for NotTerminating, but Current status is the generic "yes it exists" function that already exists in a Wait- form.

Reply:
In this particular case, RoleBindings don't have status, so the current status means less, but the user shouldn't need to know that. "Current" reconciler status is what the applier waits for. So we can expect it to be true if it synced without errors.

Reply:
Ah so is the key difference between this and the previous usage of `nt.Validate` that `nt.Validate` doesn't retry/wait?